skopt/acquisition.py

```diff
@@ -150,7 +150,7 @@ def gaussian_pi(X, model, y_opt=0.0, xi=0.01, return_grad=False):
     values[mask] = norm.cdf(scaled)
 
     if return_grad:
-        if not np.any(mask):
+        if not np.all(mask):
```

Comment:
Sorry about the confusion. `X.shape[0] > 1` raises an error before this (gradient-support is only for single samples). So this line of code is reached when there is only one sample. (Hence mask is a 1-D array of length 1 always)